build-tools/packages/build-tools/src/repoPolicyCheck/handlers/lockfiles.ts

```diff
@@ -46,4 +50,44 @@ export const handlers: Handler[] = [
             return;
         },
     },
+    {
+        name: "extraneous-lockfiles",
+        match: lockFilePattern,
+        handler: (file, root) => {
+            const manifest = getPackageManifest(root);
+            // Add the root path (.) because a lockfile is expected there
+            const knownPaths: string[] = (manifest.additionalLockfilePaths ?? []).concat(".");
+
+            // Add paths to known monorepos and packages
+            const vals = Object.values(manifest.repoPackages)
+                .filter((p) => typeof p === "string")
+                .map((p) => p.toString());
+            knownPaths.push(...vals);
+
+            // Add paths from entries that are arrays
+            const arrayVals = Object.values(manifest.repoPackages)
+                .filter((p) => typeof p !== "string")
+                .map((p) => p);
+            for (const arr of arrayVals) {
+                if (Array.isArray(arr)) {
+                    knownPaths.push(...arr.map((p) => p.toString()));
+                }
+            }
```

Comment:
The above seems compute the same thing again and again.  Should we cache it?

Reply:
Refactored this out into a get function that calculates the known paths once and returns the cached value subsequently.